### ISSUE TITLE
Feature Request: clear recorded calls on all mocks

### ISSUE THREAD

**Issue by michaelom** [mockk/mockk#153]
- [x ] I am running the latest version
- [x ] I checked the documentation and found no answer
- [x ] I checked to make sure that this issue has not already been filed

The current API comes with a handy function to unstub all mocks: `unmockkAll()`
Unfortunately there is no way to clear recorded calls in a similar fashion.

`clearMocks(...)` requires the mock to be passed as an argument. When the mock object is not in scope (e.g. test lifecycle listener) there is currently no way to clear recorded calls.

I am aware this functionality is rarely required. So is `unmockAll()` :-)

**Comment by michaelom** [comment]
Here is an example:

```
class MockResettingLifecycleListener : LifecycleListener {
    override fun afterExecuteGroup() {
         clearAllMocks()
     }
}
```

During test execution mocks reside in a (stateful) DI-Container. If not cleared after test execution, state gets carried over to the next test.

**Comment by oleksiyp** [comment]
Finally I am doing implementation lifecycle and going to take this one

**Comment by oleksiyp** [comment]
Please verify the fix in version 1.8.11 or 1.8.11.kotlin13

**Comment by michaelom** [comment]
It works.

Great work. Thanks for the new feature.
